kernel/src/snapshot.rs

```diff
@@ -422,19 +423,53 @@ mod tests {
     use object_store::path::Path;
     use object_store::ObjectStore;
 
+    use crate::actions::Format;
     use crate::engine::default::executor::tokio::TokioBackgroundExecutor;
     use crate::engine::default::filesystem::ObjectStoreFileSystemClient;
     use crate::engine::sync::SyncEngine;
     use crate::schema::StructType;
 
-    #[test]
-    fn test_snapshot_read_metadata() {
-        let path =
-            std::fs::canonicalize(PathBuf::from("./tests/data/table-with-dv-small/")).unwrap();
+    fn get_snapshot_from_path(path: &str, version: Option<Version>) -> Snapshot {
+        let path = std::fs::canonicalize(PathBuf::from(path)).unwrap();
         let url = url::Url::from_directory_path(path).unwrap();
-
         let engine = SyncEngine::new();
-        let snapshot = Snapshot::try_new(url, &engine, Some(1)).unwrap();
+        Snapshot::try_new(url, &engine, version).unwrap()
+    }
+
+    #[test]
+    fn test_replay_protocol_metadata_filtering_predicate() {
```

Comment:
So currently this just checks that we don't _break_ anything right?

I think we should probably have a more specific check in parquet reader that manually creates the expected expression and pushes it into a filter and ensures it does what it says it will.

Reply:
Cool, I'll go do that 👍

Reply:
I put up some new tests in `default/parquet.rs`. I feel like `test_parquet_protocol_metadata_filter` is a little ugly, but I don't see any easy ways to simplify it or make it more reusable.